x/ethbridge/keeper/keeper.go

```diff
@@ -104,30 +104,31 @@ func (k Keeper) ProcessBurn(ctx sdk.Context, cosmosSender sdk.AccAddress, msg *t
 	logger := k.Logger(ctx)
 	var coins sdk.Coins
 
-	if msg.Symbol == types.CethSymbol {
-		coins = sdk.NewCoins(sdk.NewCoin(types.CethSymbol, msg.CethAmount.Add(msg.Amount)))
-	} else {
-		coins = sdk.NewCoins(sdk.NewCoin(msg.Symbol, msg.Amount), sdk.NewCoin(types.CethSymbol, msg.CethAmount))
-	}
-
-	err := k.bankKeeper.SendCoinsFromAccountToModule(
-		ctx, cosmosSender, types.ModuleName, coins,
-	)
-
-	if err != nil {
-		logger.Error("failed to send coin from account to module.",
-			errorMessageKey, err.Error())
-		return err
-	}
-
 	if k.IsCethReceiverAccountSet(ctx) {
 		coins = sdk.NewCoins(sdk.NewCoin(types.CethSymbol, msg.CethAmount))
-		err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, k.GetCethReceiverAccount(ctx), coins)
+
+		err := k.bankKeeper.SendCoins(ctx, cosmosSender, k.GetCethReceiverAccount(ctx), coins)
 		if err != nil {
-			logger.Error("failed to send ceth from module to account.",
+			logger.Error("failed to send ceth from account to account.",
 				errorMessageKey, err.Error())
 			return err
 		}
+
+		coins = sdk.NewCoins(sdk.NewCoin(msg.Symbol, msg.Amount))
+
+	} else {
+		if msg.Symbol == types.CethSymbol {
+			coins = sdk.NewCoins(sdk.NewCoin(types.CethSymbol, msg.CethAmount.Add(msg.Amount)))
+		} else {
+			coins = sdk.NewCoins(sdk.NewCoin(msg.Symbol, msg.Amount), sdk.NewCoin(types.CethSymbol, msg.CethAmount))
+		}
+	}
+
+	err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, cosmosSender, types.ModuleName, coins)
```

Comment:
What will happen to these funds after they are sent to module account?

Wasn't the previous code only sending to module account first, so that it could then send to ceth receive account?

Should this send maybe be within the else statement above?

Do we have a unit test case for this?

Reply:
FYI, there was some confusion about branches, and this change doesn't belong in this PR.  Elliot's aware and will fix.

Reply:
What will happen to these funds after they are sent to module account?
there are two cases for the fund after sent to module account.
1. if the ceth receiver not set, the fund include both ceth and other token. other token will be burned, ceth kept in the module's account.
2. if the ceth receiver set, the ceth go to the receiver's account directly. other token sent to module account then burn

Wasn't the previous code only sending to module account first, so that it could then send to ceth receive account?
yes, previoud code send to module, then from module send to ceth receiver. but in cosmos sdk .42, we can't send from module to account, the receiver need some priviledge. I didn't find out interface to configure it.

Should this send maybe be within the else statement above?
it should be outside of if/else, then we will miss send other token except ceth from account to module.
then we don't have balance for burn.

Do we have a unit test case for this?
yes, we have lock/burn test case.